cdap-proto/src/main/java/co/cask/cdap/proto/metadata/lineage/ProgramInfo.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.metadata.lineage;
+
+import co.cask.cdap.proto.id.ProgramId;
+
+/**
+ * Represents the program information including when it was last executed.
+ */
+public class ProgramInfo {
+  private final ProgramId program;
```

Comment:
so we use en entity id for the program, but we use (namespace, name) for the datasets in the lineage graph. Why is that?

Reply:
no specific reason. changed to `DatasetId`.